core/metacling/test/CMakeLists.txt

```diff
@@ -25,4 +25,6 @@ ROOT_ADD_GTEST(
   TClingMethodInfoTests.cxx
 
   LIBRARIES
-  Core)
+  Core
+  Cling
```

Comment:
We must not link against `Cling`. Why is this needed?
```suggestion
```

Reply:
It is not a link-time dependency, it is a target dependency. If we change TClingCallFunc.cxx and we type `make TClingTest` we want libCling.so to be rebuilt...

Reply:
Wouldn't something like:
```
add_dependencies(TClingTest Cling)
```
be a better fit for your purpose?

Reply:
Yes, good point.